open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/test/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/lineagegraph/OpenLineageGraphValidation.java

```diff
@@ -0,0 +1,273 @@
+        }
+
+        log.debug("Validated process with qualifiedName " + process.getQualifiedName());
+    }
+
+    private void validateTable(Table table){
+        Vertex tableVertex = g.V().has(VERTEX_QUALIFIED_NAME, table.getQualifiedName())
+                .or(__.has(PROPERTY_KEY_LABEL, DATA_FILE),
+                    __.has(PROPERTY_KEY_LABEL, RELATIONAL_TABLE)).next();
+
+        assert tableVertex != null : "Table not found by qualifiedName " + table.getQualifiedName();
+
+        boolean tableIsInput = g.V(tableVertex.id()).out(TABLE_DATA_FLOW).has(PROPERTY_KEY_LABEL, PROCESS).hasNext();
+        boolean tableIsOutput = g.V(tableVertex.id()).in(TABLE_DATA_FLOW).has(PROPERTY_KEY_LABEL, PROCESS).hasNext();
+
+        if( tableIsInput && !tableIsOutput ){
+            assert processHasOutput(tableVertex.id()) : "Process missing output";
+        }
+        if( !tableIsInput && tableIsOutput ){
+            assert processHasInput(tableVertex.id()) : "Process missing input";
+        }
+        if(tableIsInput && tableIsOutput){
+            assert processHasOutput(tableVertex.id()) && processHasInput(tableVertex.id()) : "Process missing input and output" ;
+        }
+
+        log.debug("Validated table with qualifiedName " + table.getQualifiedName());
+    }
+
+    private boolean processHasOutput(Object tableId){
+        Vertex process = g.V(tableId).out(TABLE_DATA_FLOW).has(PROPERTY_KEY_LABEL, PROCESS).next();
+        return g.V(process.id()).out(TABLE_DATA_FLOW).or(__.has(PROPERTY_KEY_LABEL, RELATIONAL_TABLE),
+                        __.has(PROPERTY_KEY_LABEL, DATA_FILE)).hasNext();
+    }
+
+    private boolean processHasInput(Object tableId){
+        Vertex process = g.V(tableId).in(TABLE_DATA_FLOW).has(PROPERTY_KEY_LABEL, PROCESS).next();
+        return g.V(process.id()).in(TABLE_DATA_FLOW).or(__.has(PROPERTY_KEY_LABEL, RELATIONAL_TABLE),
+                        __.has(PROPERTY_KEY_LABEL, DATA_FILE)).hasNext();
+    }
+
+    private void validateColumn(Column column){
+        Vertex columnAsVertex = g.V().has(VERTEX_QUALIFIED_NAME, column.getQualifiedName())
+                .or(__.has(PROPERTY_KEY_LABEL, RELATIONAL_COLUMN),
+                        __.has(PROPERTY_KEY_LABEL, TABULAR_COLUMN)).next();
+
+        assert columnAsVertex != null : "Column not found by qualifiedName " + column;
+
+        if( !g.V(columnAsVertex.id()).has(PROPERTY_KEY_LABEL, TABULAR_COLUMN).in(ATTRIBUTE_FOR_SCHEMA)
+                .in("AssetSchemaType").hasLabel(DATA_FILE).hasNext() ){
+            // tabular columns that do not reach a DataFile are not eligible for validation
+            return;
+        }
+
+        boolean isInput = g.V(columnAsVertex.id()).out(COLUMN_DATA_FLOW).hasLabel("subProcess").hasNext();
+        boolean isOutput = g.V(columnAsVertex.id()).in(COLUMN_DATA_FLOW).hasLabel("subProcess").hasNext();
+
+        assert isInput || isOutput : "Column missing input and output";
```

Comment:
This check can be split into two different ones, for input/output with specific messages